test/math_errhandling_tests.c

```diff
@@ -35,6 +35,8 @@
 
 #define makelname(s) scat(s,l)
 
+volatile const int makemathname(zero_int) = 0;
```

Comment:
I think [-fno-builtin](https://gcc.gnu.org/onlinedocs/gcc/C-Dialect-Options.html#index-fno-builtin) should avoid those optimizations as well and probably makes sense in addition to making the variables volatile.

Reply:
Yeah, I want to remove -fno-builtin from the compiler flags while testing so that tests replicate what real applications are going to see. Hence the pile of volatile bits.

Reply:
And that was a silly plan -- the compilers appear to have very loose agreement with specifications around sNaN. I'm going to abandon this effort and just stick -fno-builtin back into the builds.